File: .cursor/skills/34-tietu-workflow/scripts/render_cover.mjs
#!/usr/bin/env node
/**
 * Render one 1080×1440 HTML cover with this skill's Playwright Chromium.
 *
 *   node render_cover.mjs <cover.html> <out.png>
 */
import { chromium } from 'playwright';
import fs from 'node:fs';
import path from 'node:path';
import { pathToFileURL } from 'node:url';
import { spawnSync } from 'node:child_process';

async function main() {
  const [htmlArg, outArg] = process.argv.slice(2);
  if (!htmlArg || !outArg) {
    console.error('用法: node render_cover.mjs <cover.html> <out.png>');
    process.exit(1);
  }
  const htmlFile = path.resolve(htmlArg);
  const outPng = path.resolve(outArg);
  if (!fs.existsSync(htmlFile)) {
    console.error('missing html', htmlFile);
    process.exit(1);
  }
  fs.mkdirSync(path.dirname(outPng), { recursive: true });

  const browser = await chromium.launch();
  const context = await browser.newContext({
    viewport: { width: 1080, height: 1440 },
    deviceScaleFactor: 1,
  });
  const page = await context.newPage();
  await page.goto(pathToFileURL(htmlFile).toString(), { waitUntil: 'load', timeout: 45000 });
  try {
    await page.waitForLoadState('networkidle', { timeout: 8000 });
  } catch {
    /* ignore */
  }
  await page.waitForTimeout(200);
  await page.screenshot({
    path: outPng,
    clip: { x: 0, y: 0, width: 1080, height: 1440 },
  });
  await context.close();
  await browser.close();

  const jpg = outPng.replace(/\.png$/i, '.jpg');
  spawnSync('ffmpeg', ['-y', '-i', outPng, '-q:v', '2', jpg], { stdio: 'ignore' });
  console.log('OK', fs.existsSync(jpg) ? jpg : outPng);
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: .cursor/skills/34-tietu-workflow/covers-3x4/render_cover.py
# -*- coding: utf-8 -*-
"""Render a single 3:4 HTML cover to 1080x1440 PNG/JPG via Playwright."""
from __future__ import annotations

import argparse
import subprocess
import sys
from pathlib import Path


def render_cover(html: Path, out_png: Path) -> Path:
    from playwright.sync_api import sync_playwright

    out_png.parent.mkdir(parents=True, exist_ok=True)
    with sync_playwright() as p:
        browser = p.chromium.launch()
        page = browser.new_page(viewport={"width": 1080, "height": 1440}, device_scale_factor=1)
        page.goto(html.resolve().as_uri(), wait_until="networkidle")
        page.wait_for_timeout(200)
        page.screenshot(path=str(out_png), clip={"x": 0, "y": 0, "width": 1080, "height": 1440})
        browser.close()
    jpg = out_png.with_suffix(".jpg")
    subprocess.run(
        ["ffmpeg", "-y", "-i", str(out_png), "-q:v", "2", str(jpg)],
        stdout=subprocess.DEVNULL,
        stderr=subprocess.DEVNULL,
        check=False,
    )
    print("OK", jpg if jpg.exists() else out_png)
    return out_png


def main() -> int:
    p = argparse.ArgumentParser(description="Render one 1080x1440 HTML cover")
    p.add_argument("--html", type=Path, required=True)
    p.add_argument("--out", type=Path, required=True, help="output .png path")
    args = p.parse_args()
    if not args.html.exists():
        print("missing html", args.html, file=sys.stderr)
        return 1
    render_cover(args.html, args.out)
    return 0


if __name__ == "__main__":
    raise SystemExit(main())
